lib/puma/request.rb

```diff
@@ -0,0 +1,438 @@
+# frozen_string_literal: true
+
+module Puma
```

Comment:
If this is going to be a composed object extraction, this file should be `Puma::Server::Request` and then live in `lib/puma/server/request.rb`.